src/topology/omega_complete_partial_order.lean

```diff
@@ -0,0 +1,155 @@
+/-
+Copyright (c) 2020 Simon Hudon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Simon Hudon
+-/
+import topology.basic
+import order.omega_complete_partial_order
+
+/-!
+# Scott Topological Spaces
+
+I don't know what they're good for but their notion
+of continuity is equivalent to continuity in ωCPOs.
+
+## Reference
+
+ * https://ncatlab.org/nlab/show/Scott+topology
+
+-/
+
+open omega_complete_partial_order
+open_locale classical
+
+universes u
+namespace scott_topological_space
+
+/--  -/
+def is_ωSup {α : Type u} [preorder α] (c : chain α) (x : α) : Prop :=
+(∀ i, c i ≤ x) ∧ (∀ y, (∀ i, c i ≤ y) → x ≤ y)
+
+variables (α : Type u) [omega_complete_partial_order α]
+
+/-- The characteristic function of open sets is monotone and preserves
+the limits of chains. -/
+def is_open (s : set α) : Prop :=
+continuous' (s : α → Prop)
```

Comment:
The coercion here is still breaking the abstraction. Instead use `continuous' (\lam a, a \in s)` . (Not sure if this comes up elsewhere, but the reverse coercion should be written `{x | p x}`.)

Reply:
Would it be work changing the `set` definition in core to make it `irreducible`?